src/services/resource.js

```diff
@@ -389,12 +414,88 @@ module.exports = class resourceHelper {
 		)
 
 		const commentMapping = await comments.reduce((acc, item) => {
-			acc[item.resource_id] = parseInt(item.comment_count, 10) > 0 ? true : false
+			const programIds = programResourceObj?.programResourceMapping[item.resource_id]
+				? programResourceObj?.programResourceMapping[item.resource_id]
+				: []
+			if (programIds.length > 0) {
+				programIds.forEach((programId) => {
+					acc[programId] = parseInt(item.comment_count, 10) > 0 ? true : false
+				})
+			} else {
+				acc[item.resource_id] = parseInt(item.comment_count, 10) > 0 ? true : false
+			}
+
 			return acc
 		}, {})
 		return commentMapping
 	}
 
+	/**
+	 * fetch Program Resources
+	 * @name fetchProgramResources
+	 * @param {Array} resourceIds - List of resources
+	 * @returns {Object} - Object of reourcesWithInProgram and programResourceMapping
+	 * {
+	 * 		reourcesWithInProgram : [ list of reources / solutions with in the passed program ids] ,
+	 * 		programResourceMapping : { obj of resource id with an array of program id  }
+	 *  }
+	 */
+
+	static async fetchProgramResources(resourceIds) {
+		{
+			// fetch all program details from the resource id list
+			const fetchProgramIds = await resourceQueries.findAll(
+				{
+					id: {
+						[Op.in]: resourceIds,
+					},
+					type: common.RESOURCE_TYPE_PROGRAM,
+				},
+				['id']
+			)
+			// seggregate program ids from the db response to an array
+			const programIds =
+				Array.isArray(fetchProgramIds) && fetchProgramIds.length > 0
+					? fetchProgramIds.map((program) => program.id)
+					: []
+
+			if (programIds.length > 0) {
```

Comment:
@adithyadinesh0412  add this check before
if (programIds.length === 0) {
        return { reourcesWithInProgram: [], programResourceMapping: {} };
    }